### PR TITLE
cleaned up auth and all code. also added in memory cache

### PR DESCRIPTION
**Pull request by chitalian**
1. Created in memory cache so that we can save calls made earlier within our stack (ex: proxy keys)
2. Removed prompt formatter
3. Moved all HeliconeAuth -> requestWrapper

HeliconeAuth is now just the actual raw auth values

AuthParams are the verified auth parameters generated from HeliconeAuth

### CODE REVIEW

File: worker/src/lib/providerCalls/call.ts [hunk 1]
```diff
@@ -1,4 +1,7 @@
 import { HeliconeProxyRequest } from "../HeliconeProxyRequest/mapper";
+/// <reference types="@cloudflare/workers-types" />
+import { Response } from "@cloudflare/workers-types";
+
```
Comment (chitalian) [inline worker/src/lib/providerCalls/call.ts:4]:
```suggestion
```

File: worker/src/lib/providerCalls/call.ts [hunk 2]
```diff
@@ -49,10 +49,13 @@ export async function callProvider(props: CallProps): Promise<Response> {
     const controller = new AbortController();
     const signal = controller.signal;
     setTimeout(() => controller.abort(), 1000 * 60 * 30);
-    response = await fetch(new_url.href, { ...init, signal });
+    /// <reference types="@cloudflare/workers-types" />
```
Comment (chitalian) [inline worker/src/lib/providerCalls/call.ts:52]:
```suggestion
```

File: worker/src/lib/HeliconeHeaders.ts [hunk 3]
```diff
@@ -150,13 +150,21 @@ export class HeliconeHeaders implements IHeliconeHeaders {
...
-    const heliconeAuthJWT = this.headers.get("helicone-jwt");
+
     if (heliconeAuth) {
       return {
         _type: "bearer",
-        token: heliconeAuth.replace("Bearer ", ""),
+        token: heliconeAuth,
       };
     }
+    const heliconeAuthFallback = this.headers.get("authorization");
```
Comment (colegottdank) [inline worker/src/lib/HeliconeHeaders.ts:160]: Why would this be the fallback? Is this only for api routes?

File: worker/src/lib/HeliconeProxyRequest/mapper.ts [hunk 4]
```diff
@@ -5,12 +5,7 @@ import { Result, ok } from "../../results";
 import { IHeliconeHeaders } from "../HeliconeHeaders";
```
Comment (colegottdank) [inline worker/src/lib/HeliconeProxyRequest/mapper.ts:5]: Beautiful

File: worker/src/lib/HeliconeHeaders.ts (hunk 3 above)
Comment (chitalian) [reply worker/src/lib/HeliconeHeaders.ts:160]: yep

File: worker/src/lib/dbLogger/logResponse.ts [hunk 5]
```diff
@@ -200,12 +101,12 @@ export async function logRequest(
       id: request.requestId,
       path: request.path,
       body: request.omitLog ? {} : unsupportedImage(requestBody),
-      auth_hash: request.providerApiKeyAuthHash,
+      auth_hash: "",
```
Comment (colegottdank) [inline worker/src/lib/dbLogger/logResponse.ts:104]: I'm confused

File: worker/src/lib/RequestWrapper.ts [hunk 6]
```diff
@@ -283,100 +284,103 @@ export class RequestWrapper {
...
-        data: null,
-        error: "Provider key not found",
-      };
+  if (limits.data && limits.data.length > 0) {
+    console.log("CHECKING LIMITS");
+    if (!(await checkLimits(limits.data, env))) {
+      return err("Limits are not valid");
     }
+  } else {
+    console.log("NO LIMITS");
```
Comment (colegottdank) [inline worker/src/lib/RequestWrapper.ts:358]: remove logs here

File: worker/src/lib/dbLogger/logResponse.ts (hunk 5 above)
Comment (chitalian) [reply worker/src/lib/dbLogger/logResponse.ts:104]: We should honestly remove the auth_hash column it is never used

File: worker/src/lib/RequestWrapper.ts (hunk 6 above)
Comment (chitalian) [reply worker/src/lib/RequestWrapper.ts:358]: ty

File: worker/src/lib/dbLogger/DBLoggable.ts [hunk 7]
```diff
@@ -67,18 +60,9 @@ export function dbLoggableRequestFromProxyRequest(
 ): DBLoggableProps["request"] {
   return {
     requestId: proxyRequest.requestId,
-    heliconeApiKeyAuthHash: proxyRequest.heliconeAuthHash,
```
Comment (colegottdank) [inline worker/src/lib/dbLogger/DBLoggable.ts:70]: Removed this?

**Review by colegottdank — APPROVED** [review]

**Pull request merged**